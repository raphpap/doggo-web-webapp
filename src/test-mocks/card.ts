// Types
import {Card} from 'doggo/context';

export const mockCard = (id: number): Card => ({
  attack: id,
  defense: id,
  hpLeft: id,
  hpTotal: id,
  id: `${id}`,
  image: `${id}`,
  name: `${id}`,
});
