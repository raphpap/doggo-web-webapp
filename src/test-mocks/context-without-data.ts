// Types
import {ApplicationContext, BattleStatus} from 'doggo/context';

export const mockContextWithoutData = (): ApplicationContext => ({
  actions: {
    battle: jest.fn(),
    capture: jest.fn(),
    getNextOpponent: jest.fn(),
    login: jest.fn(),
    selectBattleCard: jest.fn(),
    unselectBattleCard: jest.fn(),
  },
  state: {
    battle: {
      cardId: null,
      opponent: null,
      status: BattleStatus.NoOpponent
    },
    cards: null,
    error: null,
    loading: false
  }
});
