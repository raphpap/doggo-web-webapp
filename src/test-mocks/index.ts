export * from './card';
export * from './context-without-data';
export * from './logged-in';
