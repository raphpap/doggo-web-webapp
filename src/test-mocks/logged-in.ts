// Types
import {ApplicationContext, BattleStatus} from 'doggo/context';

// Mocks
import {mockCard, mockContextWithoutData} from '.';

export const mockLoggedIn = (): ApplicationContext => {
  const context = mockContextWithoutData();

  return {
    ...context,
    state: {
      ...context.state,
      battle: {
        ...context.state.battle,
        opponent: mockCard(99),
        status: BattleStatus.NoCardSelected
      },
      cards: [
        mockCard(1),
        mockCard(2)
      ]
    }
  };
};
