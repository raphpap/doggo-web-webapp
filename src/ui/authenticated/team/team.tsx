// Vendor
import React from 'react';
import {compose} from 'recompose';

// Vendor Types
import {RouteComponentProps, withRouter} from 'react-router';

// Assets
import logo from 'doggo/assets/images/team-logo.png';

// Context
import {
  Card,
  withApplicationContext,
  WithApplicationContextProps
} from 'doggo/context';

// Utilities
import findCard from 'doggo/utilities/find-card';

// Shared Components
import BigCardModal from 'doggo/ui/@components/big-card-modal';
import CardsContainer from 'doggo/ui/@components/cards-container';
import Logo from 'doggo/ui/@components/logo';
import Message from 'doggo/ui/@components/message';
import SmallCard from 'doggo/ui/@components/small-card';

// Types
interface Props {
  cardId?: string;
}
type EnhancedProps = Props &
  WithApplicationContextProps &
  RouteComponentProps<never>;

const enhance = compose<EnhancedProps, Props>(
  withApplicationContext,
  withRouter
);

export class Team extends React.Component<EnhancedProps> {
  public render() {
    const {cardId} = this.props;
    const {cards} = this.props.context.state;

    if (!cards) {
      return <Message>Loading...</Message>;
    }

    const card = findCard(cardId || '', cards);
    const selectedCard = card ? card : null;

    return (
      <>
        <Logo logo={logo} height={'80px'} marginUnits={0} />

        <CardsContainer>
          {cards.map((card, index) => (
            <SmallCard
              key={index}
              card={card}
              onCardClick={(card: Card) => {
                this.openModal(card.id);
              }}
            />
          ))}
        </CardsContainer>

        <BigCardModal card={selectedCard} onClose={this.closeModal} />
      </>
    );
  }

  private openModal = (cardId: string) => {
    this.props.history.replace(`/team/card/${cardId}`);
  };

  private closeModal = () => {
    this.props.history.replace('/team');
  };
}

export default enhance(Team);
