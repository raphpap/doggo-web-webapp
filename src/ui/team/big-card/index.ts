export * from './big-card';
export {default} from './big-card';
