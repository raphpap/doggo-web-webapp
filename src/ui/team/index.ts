export * from './team';
export {default} from './team';
