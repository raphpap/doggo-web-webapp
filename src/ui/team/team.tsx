// Vendor
import React from 'react';
import styled from 'react-emotion';
import {compose} from 'recompose';

// Context
import {
  withApplicationContext,
  WithApplicationContextProps
} from 'doggo-web-webapp/ui/@context';

// Components
import Card from './card';

// Elements
const CardsList = styled.ul`
  min-width: 320px;
  max-width: 800px;
  padding: 0;
`;

const Message = styled.p`
  color: rgba(255, 255, 255, 0.8);
`;

// Types
interface Props {}
type EnhancedProps = Props & WithApplicationContextProps;

const enhance = compose<EnhancedProps, Props>(withApplicationContext);

export const Team: React.SFC<EnhancedProps> = ({context}) => {
  const {state} = context;
  const {cards} = state;

  if (!cards) {
    return <Message>Loading...</Message>;
  }

  return (
    <>
      <Message>Team</Message>
      <CardsList>
        {cards.map((card, index) => (
          <Card key={index} card={card} />
        ))}
      </CardsList>
    </>
  );
};

export default enhance(Team);
