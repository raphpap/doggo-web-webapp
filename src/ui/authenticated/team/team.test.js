// Vendor
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// Components
import {Team} from './team';

// Shared Components
import BigCardModal from 'doggo/ui/@components/big-card-modal';
import SmallCard from 'doggo/ui/@components/small-card';

// Mocks
import {mockCard, mockLoggedIn} from 'doggo/test-mocks';

describe('<Team />', () => {
  let wrapper;
  let context;
  const card11 = mockCard(11);
  const card12 = mockCard(12);

  beforeEach(() => {
    context = mockLoggedIn();
    context.state.cards = [card11, card12];
    wrapper = shallow(<Team context={context} />);
  });

  it('renders a message when no cards to show', () => {
    context.state.cards = [];
    wrapper.setProps({context});
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the correct number of cards', () => {
    const smallCard = wrapper.find(SmallCard);
    expect(smallCard).toHaveLength(2);
  });

  it('shows the big card modal when a cardId is provided', () => {
    wrapper.setProps({cardId: '12'});

    const bigCardModal = wrapper.find(BigCardModal);
    expect(bigCardModal.props().card).toBe(card12);
  });

  it('doesn\'t show the big card modal when no cardId provided', () => {
    const bigCardModal = wrapper.find(BigCardModal);
    expect(bigCardModal.props().card).toBeNull();
  });
});
