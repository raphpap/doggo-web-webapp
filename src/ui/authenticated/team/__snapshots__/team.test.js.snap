// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Team /> renders a message when no cards to show 1`] = `
<Message>
  No cards to display
</Message>
`;
